esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -208,6 +208,64 @@ bool EsdlServiceImpl::loadLoggingManager(Owned<ILoggingManager>& manager, IPTree
     return true;
 }
 
+template <typename file_loader_t>
+inline bool EsdlServiceImpl::initMaskingEngineDirectory(const char* dir, const char* mask, file_loader_t loader)
+{
+    bool failed = false;
+    Owned<IDirectoryIterator> files = createDirectoryIterator(dir, mask, false, false);
+    ForEach(*files)
+    {
+        const char* filename = files->query().queryFilename();
+        Owned<IPropertyTree> ptree = loader(filename, 0, ptr_ignoreWhiteSpace, nullptr);
+        if (ptree)
+        {
+            if (!initMaskingEngineEmbedded(m_oStaticMaskingEngine, ptree, true))
+                failed = true;
+        }
+    }
+    return !failed;
+}
+
+bool EsdlServiceImpl::initMaskingEngineEmbedded(Owned<IDataMaskingEngine>& engine, const IPropertyTree* ptree, bool required)
+{
+    if (!ptree)
+        return !required;
+    bool created = false;
+    if (!engine)
+    {
+        engine.setown(new DataMasking::CEngine());
+        created = true;
+    }
+    size_t addedProfiles = engine->loadProfiles(*ptree);
+    switch (addedProfiles)
+    {
+    case size_t(-1):
+        if (created)
+            engine.clear();
+        return false;
+    case 0:
+        if (created)
+            engine.clear();
+        return !required;
+    default:
+        DBGLOG("initMaskingEngineEmbedded added %zu profiles to engine %p", addedProfiles, engine.get());
+        return true;
+    }
+}
+
+bool EsdlServiceImpl::initMaskingEngineDirectory(const char* dir)
+{
+    bool failed = false;
+    if (!isEmptyString(dir))
+    {
+        if (!initMaskingEngineDirectory(dir, "*.yaml", createPTreeFromYAMLFile))
```

Comment:
Don't necessarily need to change anything, but just a note that you should consider... most of the places I've used yaml files so far hasn't just been an alternative mark up to xml.  The format was either designed around our use of helm, kubernetes, configmaps (also kubernetes), etc.

Just consider the fact that loading them as if they were the same as your xml may mean you have to recognize another yaml format in the future coming from a different configuration/management pathway.

Reply:
One of my assumptions is that we will (eventually) support service-specific profiles. This strongly suggests that profiles will be written using xml, and may even be generated by the manifest tool. My support for yaml was intentionally as an alternative to xml.

I would like to discuss this offline, even if not changing things now.